src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.ts

```diff
@@ -67,20 +64,18 @@ export class DataFieldEditDialogComponent {
         // Add the default options for DeliveryMethod
         const defaultOptions = this.settingsService.settings.DefaultDeliveryMethodOptions;
         defaultOptions.forEach(defopt => {
-          const alreadyExists = this.data.dataFieldOptions.some(dfo =>
+          const alreadyExists = this.data.dataField.dataOptions.some(dfo =>
             dfo.optionName === defopt.optName && dfo.optionValue === defopt.optionValue);
           if (!alreadyExists) {
             const newDataOption = {
               id: uuidv4(),
-              displayOrder: this.data.dataFieldOptions.length + 1,
+              displayOrder: this.data.dataField.dataOptions.length + 1,
```

Comment:
It looks like the Reorder method in DataFieldService expects the displayOrder to be zero-based and this expects it to be one-based. Note: That Reorder is a bit strange.

Reply:
The reorder in DataFieldService reorders DataFields.  This displayOrder is for the DataOptions contained in a DataField.  Two different lists.  We can talk through the DataFieldService Reorder method if you would like to.